### PR TITLE
A85: Changes to xDS LRS Custom Metrics Support

### PR DESCRIPTION
**Pull request by markdroth**

### CODE REVIEW

**Review by ejona86 — APPROVED** [review]

File: A85-lrs-custom-metrics-changes.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+```
+
+For forward-compatibility, implementations should ignore any strings
+that do not match a currently supported ORCA field.  Implementations
+must support the following minimum set of fields:
+
+- cpu_utilization
+- mem_utilization
+- application_utilization
+- named_metrics map
```
Comment (dfawley) [inline A85-lrs-custom-metrics-changes.md:88]:
```suggestion
- `cpu_utilization`
- `mem_utilization`
- `application_utilization`
- `named_metrics map`
```

File: A85-lrs-custom-metrics-changes.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+### LRS APIs in XdsClient or LrsClient
+
+The LRS API in XdsClient (or LrsClient, if the implementation has split
+the LRS APIs into their own interface) currently provides a way to obtain
+a locality stats handle that is keyed by LRS server, CDS and EDS
+resource names, and locality.  We will add the propagation bits as
+another dimension of this key.
+
+When the xds_cluster_impl LB policy reports LRS metrics to the locality
```
Comment (dfawley) [inline A85-lrs-custom-metrics-changes.md:98]:
```suggestion
When the `xds_cluster_impl` LB policy reports LRS metrics to the locality
```

File: A85-lrs-custom-metrics-changes.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+retain this env var for at least a couple of releases as a temporary
+mechanism to get the old behavior back.  The default for the env var
+will be set to true once it passes interop tests, but users will be able
+to explicitly set it to false to get the old behavior.
+
+## Rationale
+
+Storing the propagation info inside the locality stats handle from
+the XdsClient or LrsClient is more efficient than the perhaps slightly
+more obvious solution of handling this in the xds_cluster_impl policy
```
Comment (dfawley) [inline A85-lrs-custom-metrics-changes.md:151]:
```suggestion
more obvious solution of handling this in the `xds_cluster_impl` policy
```

**Review by dfawley — APPROVED** [review]
LGTM.   A few formatting suggestions but that's it.

**Pull request merged**